pkg/pulumiyaml/codegen/eject.go

```diff
@@ -99,9 +98,9 @@ func LoadTemplate(dir string) (*workspace.Project, *ast.TemplateDecl, hcl.Diagno
 
 	main := proj.Main
 	if main == "" {
-		main = projectDir
+		main = dir
```

Comment:
I'm confused.  This is a change in behaviour, as it will now look for the `main` in a different directory than before.  What if `projectDir != dir`, it would be incorrect now, no?  

We should probably have some tests for this as well.

Reply:
Yeh this looks wrong. I suspect _dir_ is the problematic part here, something is calling LoadTemplate with the wrong argument.

Reply:
I did not know/expect that LoadTemplate would walk up the directory tree to find the project, why does it do that? Will fix the above to not change the behavior

>   I suspect dir is the problematic part here, something is calling LoadTemplate with the wrong argument.

For the issue I am working on, dir is correct (receiving `SourceDirectory` from `ConvertProgramRequest`) it is only on Windows that if `dir == projectDir` then projectDir is `.`

Reply:
Couldn't you just use `projectDir = filepath.Abs(projectDir)` before the call to `path.Dir`?  That should also solve the windows issue by just making the path an absolute path?

Reply:
Oh that's dumb, I guess we should never use `path` anywhere then :thinking: Good to know!